### PR TITLE
feat: Bedrock support added, only for Anthropic models

### PR DESCRIPTION
**Pull request by parhammmm**
Only Anthropic is supported but others can be added by adding a `Provider` class

Usage (using api keys - there are other methods):

```ts
  const bedrock = new Bedrock({
    model: BEDROCK_MODELS.ANTHROPIC_CLAUDE_3_HAIKU,
    region: "us-east-1",
    credentials: {
      accessKeyId: "...",
      secretAccessKey: "...",
    },
  });
```

### CODE REVIEW

File: packages/core/e2e/examples/nextjs-agent/next.config.mjs [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
-const nextConfig = {};
+const nextConfig = {
+  webpack: (config) => {
+    config.resolve.alias = {
+      ...config.resolve.alias,
+      "@smithy": false,
+      "@aws-sdk": false,
+    };
+    return config;
+  },
```
Comment (himself65) [inline packages/core/e2e/examples/nextjs-agent/next.config.mjs:10]: should not modify this in user side

**Review by himself65 — CHANGES_REQUESTED** [review]

File: packages/core/e2e/examples/nextjs-agent/next.config.mjs (hunk 1 above)
Comment (parhammmm) [reply packages/core/e2e/examples/nextjs-agent/next.config.mjs:10]: @himself65 I wasn't sure where it should be modified, it seems to be specific only to this example! `packages/core/src/next.ts` didn't seem like the right place, where shall I place it?

Comment (himself65) [reply packages/core/e2e/examples/nextjs-agent/next.config.mjs:10]: Yes

**Review by himself65 — CHANGES_REQUESTED** [review]
Could we move to `@llamaindex/experimental` or `@llamaindex/community`, since I'm not sure if this is stable

**Comment by parhammmm** [comment]
> Could we move to `@llamaindex/experimental` or `@llamaindex/community`, since I'm not sure if this is stable

@himself65 sure I can move it.

Which part is concerning maybe I can take a look?

**Comment by parhammmm** [comment]
> Could we move to `@llamaindex/experimental` or `@llamaindex/community`, since I'm not sure if this is stable

Done! I moved it to `@llamaindex/community`, is it's publishing to npm automatic?

**Comment by parhammmm** [comment]
@himself65 @marcusschiesser what do you guys think? let me know if any futher changes are needed

**Comment by petergoldstein** [comment]
@himself65 @marcusschiesser We'd really like to see this land.  I think @parhammmm has done everything that was requested.  Is there any additional work required to get this merged?  Thanks!

**Comment by himself65** [comment]
Sorry for the late response, I was on vacation. This is LGTM, let me do some small change

**Comment by parhammmm** [comment]
@himself65 thanks!

**Pull request merged**